src/controllers/restful/PaymentController.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * PaymentControllers
+ * RESTful Controllers for Payments Endpoints
+ */
+
+import * as express from "express";
+import { PagoPaConfig } from "../../Configuration";
+import { ControllerError } from "../../enums/ControllerError";
+import { HttpErrorStatusCode } from "../../enums/HttpErrorStatusCode";
+import * as PaymentsService from "../../services/PaymentsService";
+import { PaymentsActivationRequest } from "../../types/controllers/PaymentsActivationRequest";
+import { PaymentsCheckRequest } from "../../types/controllers/PaymentsCheckRequest";
+import * as PaymentsConverter from "../../utils/PaymentsConverter";
+import * as RestfulUtils from "../../utils/RestfulUtils";
+
+// Forward a payment check request from BackendApp to PagoPa
+export async function checkPaymentToPagoPa(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): Promise<boolean> {
+  // Validate input
+  const errorOrPaymentsCheckRequest = PaymentsCheckRequest.decode(req.params);
+  if (errorOrPaymentsCheckRequest.isLeft()) {
+    RestfulUtils.sendErrorResponse(
+      res,
+      ControllerError.ERROR_INVALID_INPUT,
+      HttpErrorStatusCode.BAD_REQUEST
+    );
+    return false;
```

Comment:
can we return an `Either` instead of `false` ? (ie, in this case `return errorOrPaymentsCheckRequest`). In this way we let erros "bubble up" so the caller is always informed about the reason af any error (useful during tests as well). 
when you return false the reason is 'swallowed'